### PR TITLE
Various logging improvements

### PR DESCRIPTION
**Pull request by tobim**
This encompasses 3 changes:
* UUIDs are consistently lowercase in log messages
* The local segment store appends the partition id to its actor ids. (This required an API change for store plugins).
* The partition transformer is logged with actor IDs now like all other actors.

### :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [ ] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: libvast/include/vast/uuid.hpp
Comment (dominiklohmann) [inline libvast/include/vast/uuid.hpp:110]: This comment is now outdated.

File: libvast/src/segment_store.cpp
Comment (dominiklohmann) [inline libvast/src/segment_store.cpp:485]: It seems very risky to me to make this formatter change in a PR that I assume you intended to merge during the feature freeze. We need to carefully test this with older databases, or merge this only after the release.

**Review by dominiklohmann — CHANGES_REQUESTED** [review]

File: libvast/include/vast/plugin.hpp
Comment (tobim) [reply libvast/include/vast/plugin.hpp:280]: I reverted this change an now retrieve the id from the segment.

File: libvast/src/segment_store.cpp
Comment (tobim) [reply libvast/src/segment_store.cpp:485]: Let's wait until after the release then.

File: libvast/native-plugins/segment_store.cpp
Comment (dominiklohmann) [inline libvast/native-plugins/segment_store.cpp:190]:
```suggestion
  self->state.name = "passive-store";
```

**Review by dominiklohmann — APPROVED** [review]
I'm all for merging this quickly after the release.

**Comment by tobim** [comment]
[sc-32366]

File: libvast/src/segment_store.cpp
Comment (dominiklohmann) [reply libvast/src/segment_store.cpp:485]: I think we can rebase and merge this.

**Pull request merged**
